src/app/components/main/main.component.ts

```diff
@@ -1,31 +1,34 @@
-import { Component, OnInit } from '@angular/core';
-// import {GeoDataService} from '../../services/geo-data.service';
-// import {Feature} from '../../models/models';
+import { Component, AfterContentInit } from '@angular/core';
 import {AuthenticatedUser, AuthService} from '../../services/authentication.service';
-import {Observable} from 'rxjs';
-import {BsModalRef, BsModalService} from 'ngx-foundation';
-import {ModalFileBrowserComponent} from '../modal-file-browser/modal-file-browser.component';
-import {RemoteFile} from 'ng-tapis';
+import {MAIN, LOGIN} from '../../constants/routes';
 import { Router, ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'app-main',
   templateUrl: './main.component.html',
   styleUrls: ['./main.component.styl']
 })
-export class MainComponent implements OnInit {
+export class MainComponent implements AfterContentInit {
   public currentUser: AuthenticatedUser;
+  private afterLoginRoute: string;
 
   constructor(
     private route: ActivatedRoute,
     private router: Router,
-    private authService: AuthService, private bsModalService: BsModalService,) {}
+    private authService: AuthService) {
+    // should support redirect to public project on login and offer way to switch to private view if user is
+    // project member (see https://jira.tacc.utexas.edu/browse/DES-1921)
+    this.afterLoginRoute = MAIN;
+  }
 
-  ngOnInit() {
+  ngAfterContentInit() {
+    if (this.authService.isLoggedIn()) {
+      this.authService.getUserInfo();
+    }
     this.authService.currentUser.subscribe(next => this.currentUser = next);
```

Comment:
`this.authService.currentUser.subscribe(next => this.currentUser = next ? next : undefined);`
This gets rid of the exception on logout